docs/decisions/adr-extend-application-manifest.md

```diff
@@ -1,79 +1,120 @@
+## Considered Options
 
-<!-- This is an optional element. Feel free to remove. -->
-## Pros and Cons of the Options
+  * For the converged application manifest format only one option is considered:
 
-### {title of option 1}
+      1. Creation of a new API endpoint delivered the manifest in the desired final format. This
+      	 moves the responsibility for the generation of the format from the clients to the Epinio
+      	 server, and trivially to the desired outcome of having all clients export the same format.
 
-<!-- This is an optional element. Feel free to remove. -->
-{example | description | pointer to more information | …}
+  * For the persistence of the new application information two options were considered:
 
-* Good, because {argument a}
-* Good, because {argument b}
-<!-- use "neutral" if the given argument weights neither for good nor bad -->
-* Neutral, because {argument c}
-* Bad, because {argument d}
-* … <!-- numbers of pros and cons can vary -->
+      1. Saving to a new adjunct kubernetes secret resource.
+      2. Extending the application CRD with fields for the new application information.
 
-### {title of other option}
+## Decision Outcome
 
-{example | description | pointer to more information | …}
+We are extending the application CRD with fields for the new application information.
 
-* Good, because {argument a}
-* Good, because {argument b}
-* Neutral, because {argument c}
-* Bad, because {argument d}
-* …
+While both persistence options will lead to stable persistence the chosen option is believed to be
+likely less complex, as only the code reading and writing application CR's has to be adapted to
+(extended for) the new information. Whereas with a separate resource completely new code will be
+required, not just for reading and writing, but also construction and destruction.
 
-<!-- This is an optional element. Feel free to remove. -->
-## More Information
+## Validation
 
-{You might want to provide additional evidence/confidence for the decision outcome here and/or
- document the team agreement on the decision and/or
- define when and how this decision should be realized and if/when it should be re-visited and/or
- how the decision is validated.
- Links to other decisions and resources might appear here as well.}
\ No newline at end of file
+The implementation of both API endpoint and extended application information will be validated
+through additional API and command tests added to the testsuites used by the CI/CD system.
+
+## Specification Details
+
+### API endpoint
+
+The new API endpoint is an extension of the existing `AppPart` endpoint, i.e. of
+`/namespaces/:namespace/applications/:app/part/:part`.
+
+This endpoint is extended with a new allowed part value `manifest`, which, when requested delivers a
+`text/plain` result containing the application manifest serialized into its final form.
```

Comment:
A request of `accept: application/yaml` can return a `content-type: application/yaml`. Those might be more applicable header fields.

Reply:
The existing code for part `values` (which is YAML data as well) is https://github.com/epinio/epinio/blob/3d118b353f9501c87052ac93dede24de385f6174/internal/api/v1/application/part.go#L316-L324

My plan was to implement the new part in the same way.

Actually, I suspect that this code returns an `application/octet-stream`. ... Verified.

Reply:
I will change to ADR to say `application/octet-stream` and reference the existing part value.
The Web UI already handles part `values` correctly, right ?

Reply:
Done in commit bd63277117ad6203e03fbc946024de8a6079b62d

Reply:
`application/octet-stream` is normally used for an unknown bag of binary data. Here though we know it's yaml, so `application/yaml` is more appropriate.

We do use `/part` to fetch `values`, `chart` and `image`. For `values` we request `text/plain` otherwise `blob`. The former is ignored by the API, and the UI ignores whatever the response type is. So nothing will break with the existing or proposed approach. I don't think this is blocking, but it might be worth at some point updating `/parts` to return `content-type` aligned with response.

Reply:
I agree that we should be looking at this again in the future for better content-type.